pokecrystalvc.patch.template

```diff
@@ -0,0 +1,734 @@
+;    011251
+;    
+[FPA 007 Begin]                                                   
+Mode = 3                                                          
+Type = 0                                                          
+Address = {{address}}   
+DarkEnable0 = 1
+Dark0 = 5   
+MotionBEnable0 = 3                                            
+MotionBlur0 = 7                                              
+ConditionType = 0                                                 
+ConditionValueA = a2: c2 cf  
+ConditionValueB = a2: 00 00
+ConditionValueC = a2: {{constant:constants/move_constants.asm:HYPER_BEAM}} 00                                                          
+
+
+                        
+                                                                                                                            
+;-----111111111111111144444444444444----0xc902 no ..............Mem Write: pc32 = 0xcc46a addr = 0xc902 value = 0xd                              
+                                      
+;000cc473h: FE 4F D0 cd                                        ; 﨩?    
+;000cc495h: FE 4F D0 CD                                     ; 﨩型
+;000cc497h: FE 4F D0 CD                                     ; 﨩型
+; -------------0xd4170xd4170xd4170xd417---------------   Mem Write: pc32 = 0x3a89 addr = 0xd417 value = 0xd1
+;000cc486h: FE 4F D0 CD 7D 3A 22 FA 19 D4 22 C5 E5 6F 26 00 ; 﨩型}:"??佩o&.                                                                                                                                                    
+[FPA 042801 Begin]                                                                                                                               
+Mode = 3                                                                                                                                         
+Type = 0                                                                                                                                         
+Address = {{Address}}                                                                                                                       
+DarkEnable0 = 1                                                                                                                                  
+Dark0 = 5                                                                                                                                        
+MotionBEnable0 = 3                                                                                                                               
+MotionBlur0 = 11                                                                                                                                 
+ConditionType = 0                                                                                                                                
+ConditionValueA = a4: c2 cf 17 d4                                                                              
```

Comment:
Just to make sure this task is listed in the todo:
ConditionValueA is a hex string containing pairs of addresses to compare.
This string consists of the following two addresses:
```
00:cfc2 wFXAnimID
05:d417 wBattleAnimByte
```

To keep things simple for now, I suggest having a command that expands an address into two hexadecimal values, so something like `{{CondAddress:wFXAnimID}}` would expand into `C2 CF` (And a lowercase variant would work similarly).

Also use the correct constant values for the values this is compared against in ConditionValueC. wBattleAnimByte probably contains a battle anim command.

I'm not sure what ConditionValueB means yet, but it might be the comparison type (0 meaning equality)

Reply:
Starting off... Perhaps @mid-kid or @Rangi42 could help me understand this a bit better. How exactly does it know which bank it needs to look in, if its not provided? I hoping this is more of a generic assembly specific question.

Reply:
It's very likely it just looks in the current wram bank whenever the hook is reached. Just like with the rom, bank 0 is always loaded, the rest depends on the current state.

If there was a way to explicitly specify the bank I'm sure we would've noticed it. A 5 instead of a 0 is a bit hard to miss.

Reply:
Added `ConditionValueB` commands and updated `.patch.template` file and `make_patch.c`. See https://github.com/pret/pokecrystal/pull/882/commits/dfc143764abbbd308204eda4a7ca2f461651f905

Reply:
Still need to implement ConditionValueC constants where known. This will require some rework on how we obtain values for constants.

Reply:
We discussed a solution in #pokecrystal involving a file that just does `INCLUDE "constants.asm"` followed by printing out all the necessary constants, sym-file style; and use `rgbasm > $(mktemp)` to capture its output.